apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js

```diff
@@ -679,6 +680,31 @@ function getCloseButtonUrl( calypsoPort ) {
 	};
 }
 
+/**
+ * Ensures gutenboarding status and corresponding data is placed on the calypsoifyGutenberg object.
+ * This is imporant because it allows us to adapt small changes to the editor when
+ * used in the context of Gutenboarding.
+ *
+ * @param {MessagePort} calypsoPort Port used for communication with parent frame.
+ */
+function getGutenboardingStatus( calypsoPort ) {
+	const { port1, port2 } = new MessageChannel();
+	calypsoPort.postMessage(
+		{
+			action: 'getGutenboardingStatus',
+			payload: {},
+		},
+		[ port2 ]
+	);
+	port1.onmessage = ( { data } ) => {
+		const { isGutenboarding, frankenflowUrl } = data;
+		calypsoifyGutenberg.isGutenboarding = isGutenboarding;
+		calypsoifyGutenberg.frankenflowUrl = frankenflowUrl;
+		// Hook necessary if message recieved after editor has loaded.
+		window.wp.hooks.doAction( 'updateLaunchButton', isGutenboarding );
```

Comment:
Do you think it makes sense to switch to creating an @wordpress/data store? Then, it could update state through that instead of the option.

Theoretically, I'd love to switch the FSE button to the same mechanism if possible. It might be more effort than we need to put in at this point, but it might be useful if there is more data we want to think about for the state here

Reply:
Thats about what I thought as well.  The @wordpress/data store would work well, but since we have the calypsoifyGutenberg object available and only need to tack on two static values it didn't seem worth setting up yet.  That could definitely be something worth doing in the end though, especially if we end up needing to handle more state.

Reply:
Yeah, I agree with @Addison-Stavlo on this one - I ran into a similar thing with the background color PR (#39045). I think once we get further along with frankenflow, we'll have something in the store to reference, opposed to adding an arbitrary one now.

Reply:
That makes a ton of sense! :)